upi/vsphere/vm/ignition.tf

```diff
@@ -0,0 +1,76 @@
+locals {
+  mask = element(split("/", var.machine_cidr), 1)
+  gw   = cidrhost(var.machine_cidr, 1)
+  dns_addresses_rendered = join(
+    "\n",
+    formatlist(
+      "%s=%s",
+      null_resource.generate_nm_dns_addresses.*.triggers.dns_item,
+      var.dns_addresses,
+    ),
+  )
+
+  ignition_encoded = "data:text/plain;charset=utf-8;base64,${base64encode(var.ignition)}"
+}
+
+/* generate_nm_dns_addresses just creates
+ * DNS1 and DNS2 to be
+ * used in local dns_addresses_rendered
+ */
+resource "null_resource" "generate_nm_dns_addresses" {
+  count = length(var.dns_addresses)
+
+  triggers = {
+    dns_item = "DNS${count.index + 1}"
+  }
+}
+
+data "ignition_file" "hostname" {
+  count = length(var.fqd_names)
+
+  filesystem = "root"
+  path       = "/etc/hostname"
+  mode       = "420"
+
+  content {
+    content = element(split(".", var.fqd_names[count.index]), 0)
+  }
+}
+
+data "ignition_file" "static_ip" {
+  count = length(var.ip_addresses)
+
+  filesystem = "root"
+  path       = "/etc/sysconfig/network-scripts/ifcfg-ens192"
+  mode       = "420"
+
+  content {
+    content = <<EOF
+TYPE=Ethernet
+BOOTPROTO=none
+NAME=ens192
+DEVICE=ens192
+ONBOOT=yes
+IPADDR=${var.ip_addresses[count.index]}
+PREFIX=${local.mask}
+GATEWAY=${local.gw}
+DOMAIN=${var.cluster_domain}
+${local.dns_addresses_rendered}
```

Comment:
why not https://www.terraform.io/docs/configuration/expressions.html#string-templates

Reply:
I just took what was converted from 0.11.